### PR TITLE
[596] Add Items in Interconnection View

### PR DESCRIPTION
**Pull request by gdaniel**
Bug: https://github.com/eclipse-syson/syson/issues/596

This PR also adds support for Flows involving items. This is not in a separate commit because adding items required to fix the existing edges to make sure there is no tool that doesn't work. Doing so also added the support for items in flow.

Ticket #588 can also be closed once this PR is merged.

### CODE REVIEW

File: backend/views/syson-diagram-interconnection-view/src/main/resources/images/Feature_InOut.svg
Comment (AxelRICHARD) [inline backend/views/syson-diagram-interconnection-view/src/main/resources/images/Feature_InOut.svg:1]: Can you rename this file to _Feature_Inout.svg_ to be consistent with other svg with same pattern please?

File: backend/views/syson-diagram-interconnection-view/src/main/java/org/eclipse/syson/diagram/interconnection/view/nodes/ItemUsageBorderNodeDescriptionProvider.java
Comment (AxelRICHARD) [inline backend/views/syson-diagram-interconnection-view/src/main/java/org/eclipse/syson/diagram/interconnection/view/nodes/ItemUsageBorderNodeDescriptionProvider.java:1]: Unless I'm missing something, Items should also be available as BorderNodes for Parts

Comment (gdaniel) [reply backend/views/syson-diagram-interconnection-view/src/main/java/org/eclipse/syson/diagram/interconnection/view/nodes/ItemUsageBorderNodeDescriptionProvider.java:1]: The specification doesn't define `param` for parts (and its super types). It defines `ports`, which I assume are strictly ports and not items
image

Comment (AxelRICHARD) [reply backend/views/syson-diagram-interconnection-view/src/main/java/org/eclipse/syson/diagram/interconnection/view/nodes/ItemUsageBorderNodeDescriptionProvider.java:1]: Ok

File: backend/views/syson-diagram-interconnection-view/src/main/resources/images/Feature_InOut.svg
Comment (gdaniel) [reply backend/views/syson-diagram-interconnection-view/src/main/resources/images/Feature_InOut.svg:1]: done

**Review by AxelRICHARD — APPROVED** [review]

**Comment by AxelRICHARD** [comment]
Thank you for your contribution.
I will rebase the PR and handle the conflict.

**Comment by AxelRICHARD** [comment]
> Thank you for your contribution. I will rebase the PR and handle the conflict.

Great, you did it, thank you.

**Pull request merged**